application/src/main/java/org/thingsboard/server/controller/AlarmCommentController.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.web.bind.annotation.RestController;
+import org.thingsboard.server.common.data.alarm.AlarmComment;
+import org.thingsboard.server.common.data.exception.ThingsboardException;
+import org.thingsboard.server.common.data.id.AlarmCommentId;
+import org.thingsboard.server.common.data.id.AlarmId;
+import org.thingsboard.server.common.data.page.PageData;
+import org.thingsboard.server.common.data.page.PageLink;
+import org.thingsboard.server.queue.util.TbCoreComponent;
+import org.thingsboard.server.service.entitiy.alarm.TbAlarmCommentService;
+import org.thingsboard.server.service.security.permission.Operation;
+import org.thingsboard.server.service.security.permission.Resource;
+
+import static org.thingsboard.server.controller.ControllerConstants.ALARM_ID_PARAM_DESCRIPTION;
+import static org.thingsboard.server.controller.ControllerConstants.PAGE_DATA_PARAMETERS;
+import static org.thingsboard.server.controller.ControllerConstants.PAGE_NUMBER_DESCRIPTION;
+import static org.thingsboard.server.controller.ControllerConstants.PAGE_SIZE_DESCRIPTION;
+import static org.thingsboard.server.controller.ControllerConstants.TENANT_OR_CUSTOMER_AUTHORITY_PARAGRAPH;
+import static org.thingsboard.server.controller.ControllerConstants.UUID_WIKI_LINK;
+
+@RestController
+@TbCoreComponent
+@RequiredArgsConstructor
+@RequestMapping("/api")
+public class AlarmCommentController extends BaseController {
+    public static final String ALARM_ID = "alarmId";
+    public static final String ALARM_COMMENT_ID = "commentId";
+
+    private final TbAlarmCommentService tbAlarmCommentService;
+
+    @ApiOperation(value = "Create or update Alarm Comment ",
+            notes = "Creates or Updates the Alarm Comment. " +
+                    "When creating comment, platform generates Alarm Comment Id as " + UUID_WIKI_LINK +
+                    "The newly created Alarm Comment id will be present in the response. Specify existing Alarm Comment id to update the alarm. " +
+                    "Referencing non-existing Alarm Comment Id will cause 'Not Found' error. " +
+                    "\nRemove 'id' and optionally 'userId' from the request body example (below) to create new Alarm comment entity. " +
+                    TENANT_OR_CUSTOMER_AUTHORITY_PARAGRAPH
+            , produces = MediaType.APPLICATION_JSON_VALUE)
+    @PreAuthorize("hasAnyAuthority('TENANT_ADMIN', 'CUSTOMER_USER')")
+    @RequestMapping(value = "/alarm/{alarmId}/comment", method = RequestMethod.POST)
+    @ResponseBody
+    public AlarmComment saveAlarmComment(@ApiParam(value = ALARM_ID_PARAM_DESCRIPTION)
+                                         @PathVariable(ALARM_ID) String strAlarmId, @ApiParam(value = "A JSON value representing the comment.") @RequestBody AlarmComment alarmComment) throws ThingsboardException {
+        checkParameter(ALARM_ID, strAlarmId);
+        alarmComment.setTenantId(getTenantId());
+        alarmComment.setAlarmId(new AlarmId(toUUID(strAlarmId)));
```

Comment:
We need to check that the specified alarm exists and we have access to the alarm. Something like:

```
checkAlarmId(alarmId, Operation.WRITE);
```